test/functional/rpc_packages.py

```diff
@@ -377,6 +377,9 @@ def test_submitpackage(self):
         self.log.info("Submitpackage only allows packages of 1 child with its parents")
         # Chain of 3 transactions has too many generations
         legacy_pool = node.getrawmempool()
+
+        assert_raises_rpc_error(-22, "TX decode failed:", node.submitpackage, ["", ""])
```

Comment:
I don't think just passing two empty strings is the right way of testing it because:
- it only plays with non-hex values and do not exercise the hex parsing.
- if the first tx is invalid, the second one has no effect.

Reply:
Ya that makes sense!

What I did in [34c6643](https://github.com/bitcoin/bitcoin/pull/31139/commits/34c6643c2fe0e73d73edb2ddddf0900e4256de57)

Is I used a proper hex value and then had the second value malformed so that we can cover more bases

let me know if that looks good to you I can modify it other ways aswell!

Reply:
> it only plays with non-hex values and do not exercise the hex parsing.

an empty string is valid hex, it's just an invalid tx, no? If you want to test invalid hex that's another thing.

> if the first tx is invalid, the second one has no effect.

You need two transactions currently for it to get far enough to try and deserialize. I don't really care which one(s) are valid except for ease of reading.

> let me know if that looks good to you I can modify it other ways aswell!

imo having valid hex is a more realistic scenario, but coverage is coverage

Reply:
would be good to double-check that the first tx doesn't make it into the mempool even if it otherwise would

Reply:
Agree, having this assertion would be good because the valid tx would still not make to mempool in this case.

Reply:
ok I added an assert to check for `txid` in `getrawmempool` in [1fab1d6](https://github.com/bitcoin/bitcoin/pull/31139/commits/1fab1d601e40edc7296cf3d60a7e917b0e6493b3)

let me know if that looks good to you!